core/http-auth/src/test/java/software/amazon/awssdk/http/auth/AwsV4QueryHttpSignerTest.java

```diff
@@ -0,0 +1,105 @@
+            SyncSignRequest.builder(AwsCredentialsIdentity.create("access", "secret"))
+        );
+
+        SyncSignedRequest signedRequest = signer.sign(request);
+
+        assertEquals(expectedAmzSignature, signedRequest.request().rawQueryParameters().get("X-Amz-Signature").get(0));
+        assertEquals(expectedAmzCredentials, signedRequest.request().rawQueryParameters().get("X-Amz-Credential").get(0));
+        assertEquals(expectedAmzDateHeader, signedRequest.request().rawQueryParameters().get("X-Amz-Date").get(0));
+        assertEquals(expectedAmzExpires, signedRequest.request().rawQueryParameters().get("X-Amz-Expires").get(0));
+    }
+
+    @Test
+    public void sign_withInvalidExpiration_shouldThrow() {
+        SyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicRequest(
+            SdkHttpRequest.builder(),
+            SyncSignRequest.builder(AwsCredentialsIdentity.create("access", "secret"))
+                .putProperty(EXPIRATION_INSTANT, Instant.MAX)
+        );
+
+        RuntimeException exception = assertThrows(RuntimeException.class, () -> signer.sign(request));
+
+        assertThat(exception.getMessage()).contains("valid for at most 7 days");
+    }
+
+    /**
+     * Tests that if passed anonymous credentials, signer will not generate a signature.
+     */
+    @Test
+    public void sign_withAnonymousCredentials_shouldNotSign() {
+        SyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicRequest(
+            SdkHttpRequest.builder(),
+            SyncSignRequest.builder(new TestUtils.AnonymousCredentialsIdentity())
+        );
+
+        SyncSignedRequest signedRequest = signer.sign(request);
+
+        assertEquals(signedRequest.request().rawQueryParameters().size(), 0);
+    }
+
+    @Test
+    public void signAsync_withBasicRequest_shouldSign() {
+        final String expectedAmzSignature = "bf7ae1c2f266d347e290a2aee7b126d38b8a695149d003b9fab2ed1eb6d6ebda";
+        final String expectedAmzCredentials = "access/19810216/us-east-1/demo/aws4_request";
+        final String expectedAmzDateHeader = "19810216T063000Z";
+        final String expectedAmzExpires = "604800";
+
+        AsyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicAsyncRequest(
+            SdkHttpRequest.builder(),
+            AsyncSignRequest.builder(AwsCredentialsIdentity.create("access", "secret"))
+        );
+
+        AsyncSignedRequest signedRequest = signer.signAsync(request);
+
+        assertEquals(expectedAmzSignature, signedRequest.request().rawQueryParameters().get("X-Amz-Signature").get(0));
+        assertEquals(expectedAmzCredentials, signedRequest.request().rawQueryParameters().get("X-Amz-Credential").get(0));
+        assertEquals(expectedAmzDateHeader, signedRequest.request().rawQueryParameters().get("X-Amz-Date").get(0));
+        assertEquals(expectedAmzExpires, signedRequest.request().rawQueryParameters().get("X-Amz-Expires").get(0));
```

Comment:
seems like we are missing asserts on the Algorithm and SignedHeaders query params

Reply:
I'm pretty sure those are implicit in the signature, but I can add explicit checks

Reply:
True for the other query params asserted here too, but for all of these, the asserts become documentation that helps understand the expected behavior better.